cpp/src/arrow/filesystem/hdfs.cc

```diff
@@ -46,7 +46,7 @@ using internal::GetAbstractPathParent;
 using internal::MakeAbstractPathRelative;
 using internal::RemoveLeadingSlash;
 
-static constexpr int32_t kDefaultHdfsPort = 8020;
+static constexpr int32_t kDefaultHdfsPort = 0;
```

Comment:
Does this fix mean the port number will be chosen according to the URI scheme (hdfs / viewfs)? If so, perhaps add a comment to that effect, so that we don't break the feature by mistake again.

Reply:
Yes, this actually the default:
https://github.com/apache/hadoop/blob/trunk/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java#L383

handled here:
https://github.com/apache/hadoop/blob/trunk/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java#L362

i.e. 8020 will be set for DistributedFileSystem:
https://github.com/apache/hadoop/blob/e346e3638c595a512cd582739ff51fb64c3b4950/hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DistributedFileSystem.java#L1816

Actually maybe the best is to remove this constant as it is misleading.